### PR TITLE
Add hook to fix reordered samples

### PR DESCRIPTION
**Pull request by PJungkamp**

### CODE REVIEW

File: lib/hooks/reorder_ts.cpp [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+				if (i != window.size() - 1) {
+					logger->warn("Fixing reordered Sample");
+					sample_dump(logger, sample);
+				}
+
+				if (window.size() == window_size) {
+					// The front sample will be returned.
+					Sample *window_sample = window.front();
+					// Move all elements before the index of insertion towards the front.
+					std::memmove(&window[0], &window[1], i * sizeof(decltype(window)::value_type));
```
Comment (stv0g) [inline lib/hooks/reorder_ts.cpp:121]: Is there a reason you are using `std::memmove` here? Would't `std::vector::insert()` be easier to understand?

**Review by stv0g — CHANGES_REQUESTED** [review]

File: lib/hooks/reorder_ts.cpp (hunk 1 above)
Comment (PJungkamp) [reply lib/hooks/reorder_ts.cpp:121]: Using `insert` with iterators into `this` is undefined behaviour. The `std::copy` and `std::copy_backward` algorithms might work, but seemed similar in verbosity.

File: lib/hooks/reorder_ts.cpp [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+namespace node {
+
+class ReorderTsHook : public Hook {
+
+protected:
+	std::vector<Sample *> window;
+	std::size_t window_size;
+	Sample *buffer;
+
+	void swapSample(Sample *lhs, Sample *rhs) {
```
Comment (stv0g) [inline lib/hooks/reorder_ts.cpp:28]:
```suggestion
	void swapSample(Sample *lhs, Sample *rhs)
	{
```

**Review by stv0g — APPROVED** [review]

File: lib/hooks/reorder_ts.cpp [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+			window.push_back(sample_clone(smp));
+
+			logger->debug("window.size={}/{}", window.size(), window_size);
+
+			return Hook::Reason::SKIP_SAMPLE;
+		}
+
+		for (std::size_t i = window.size() - 1;; i--) {
+			if (time_cmp(&smp->ts.origin, &window[i]->ts.origin) >= 0) {
+				if (i != window.size() - 1) {
```
Comment (stv0g) [inline lib/hooks/reorder_ts.cpp:106]: Please remove `{}` for single statements

**Review by stv0g — APPROVED** [review]
Looks good :) I think we can merge this now.

Can you address the last comment which I added and also add the missing OpenAPI spec for the hook config?

Thanks :)

**Comment by PJungkamp** [comment]
- [x] Added OpenAPI specification.
- [x] Fixed `if` braces.

**Review by stv0g — APPROVED** [review]

**Pull request merged**